zksync_sdk/zksync_provider/batch_builder.py

```diff
@@ -0,0 +1,578 @@
+
+    def __init__(self, wallet: Wallet, nonce: int, txs: List[EncodedTx] = None):
+        if txs is None:
+            txs = []
+        self.wallet = wallet
+        self.nonce = nonce
+        self.transactions: List[dict] = []
+        for tx in txs:
+            value = tx.dict()
+            value[self.IS_ENCODED_TRANSACTION] = True
+            value[self.ENCODED_TRANSACTION_TYPE] = tx.tx_type()
+            self.transactions.append(value)
+
+    async def build(self) -> BatchResult:
+        if not self.transactions:
+            raise RuntimeError("Transaction batch cannot be empty")
+        res = await self._process_transactions()
+        trans = res["trans"]
+        signature = self.wallet.eth_signer.sign(res["msg"].encode())
+        return BatchResult(trans, signature, res["total_fee"])
+
+    def add_withdraw(self,
+                     eth_address: str,
+                     token: TokenLike,
+                     amount: Decimal,
+                     fee: Decimal = None,
+                     valid_from=DEFAULT_VALID_FROM,
+                     valid_until=DEFAULT_VALID_UNTIL
+                     ):
+        withdraw = {
+            self.ENCODED_TRANSACTION_TYPE: EncodedTxType.WITHDRAW,
+            self.IS_ENCODED_TRANSACTION: False,
+            "eth_address": eth_address,
+            "token": token,
+            "amount": amount,
+            "fee": fee,
+            "valid_from": valid_from,
+            "valid_until": valid_until
+        }
+        self.transactions.append(withdraw)
+
+    def add_mint_nft(self,
+                     content_hash: str,
+                     recipient: str,
+                     fee_token: TokenLike,
+                     fee: Decimal = None
+                     ):
+        mint_nft = {
+            self.ENCODED_TRANSACTION_TYPE: EncodedTxType.MINT_NFT,
+            self.IS_ENCODED_TRANSACTION: False,
+            "content_hash": content_hash,
+            "recipient": recipient,
+            "fee_token": fee_token,
+            "fee": fee
+        }
+        self.transactions.append(mint_nft)
+
+    def add_withdraw_nft(self,
+                         to_address: str,
+                         nft_token: NFT,
+                         fee_token: TokenLike,
+                         fee: Decimal = None,
+                         valid_from=DEFAULT_VALID_FROM,
+                         valid_until=DEFAULT_VALID_UNTIL
+                         ):
+        withdraw_nft = {
+            self.ENCODED_TRANSACTION_TYPE: EncodedTxType.WITHDRAW_NFT,
+            self.IS_ENCODED_TRANSACTION: False,
+            "to_address": to_address,
+            "nft_token": nft_token,
+            "fee_token": fee_token,
+            "fee": fee,
+            "valid_from": valid_from,
+            "valid_until": valid_until
+        }
+        self.transactions.append(withdraw_nft)
+
+    # INFO: In discussion it has been figured out that SWAP operation can't be implemented in batch for now
```

Comment:
Why? In JS SDK is was possible https://github.com/matter-labs/zksync/blob/f59e154865374bdc0f5ded2e2604dac599cb75ee/sdk/zksync.js/src/batch-builder.ts#L189

Reply:
ok. Let me recheck, because I had the discusstion and I was told that it's not supported on the Server side right now.